kyo-scheduler/jvm/src/main/scala/kyo/scheduler/InternalTimer.scala

```diff
@@ -0,0 +1,31 @@
+package kyo.scheduler
+
+import InternalTimer.*
+import java.util.concurrent.ScheduledExecutorService
+import java.util.concurrent.TimeUnit
+import scala.concurrent.duration.Duration
+
+abstract private class InternalTimer:
+    def schedule(interval: Duration)(f: => Unit): TimerTask
+    def scheduleOnce(delay: Duration)(f: => Unit): TimerTask
+
+private[kyo] object InternalTimer:
+
+    abstract class TimerTask:
+        def cancel(): Boolean
+
+    def apply(executor: ScheduledExecutorService): InternalTimer =
+        new InternalTimer:
+            def schedule(interval: Duration)(f: => Unit): TimerTask =
+                val future = executor.scheduleWithFixedDelay(() => f, interval.toNanos, interval.toNanos, TimeUnit.NANOSECONDS)
+                new TimerTask:
+                    def cancel() = future.cancel(true)
+            end schedule
+
+            def scheduleOnce(delay: Duration)(f: => Unit): TimerTask =
+                val future = executor.schedule((() => f): Runnable, delay.toNanos, TimeUnit.NANOSECONDS)
+                new TimerTask:
```

Comment:
Is it possible to avoid this allocation?

Reply:
You're getting used to Kyo's focus on performance :) In this case, I don't think it's necessary because only two tasks are submitted when the scheduler starts. It'd also be more difficult to mock this class in tests if we used an opaque type to alias `ScheduledFuture`.